### PR TITLE
Update ESLint packages and adjust to new rules

### PR DESCRIPTION
**Pull request by jbalboni**
Connects to https://github.com/department-of-veterans-affairs/vets.gov-team/issues/4363

This updates the ESLint packages to the latest, except for the a11y rules, which are a version back because that's what the latest airbnb config requires.

I've tried to make only small changes, and turn off rules with big impacts. Still, it's a lot of changes. My hope is to turn on some or all of the rules that are newly turned off in our config, per discussion with the team.

I've also added comments to ignore some rules in certain files, if the rules seemed like ones we obviously want on, but were too difficult to fix in some instances. This mostly affects a11y rules.

If you're curious, this is the frequency of those turned off errors:

screen shot 2017-08-17 at 2 27 02 pm

### CODE REVIEW

File: src/js/pensions/config/form.js [hunk 1]
```diff
@@ -110,7 +110,8 @@ const {
 const nonRequiredFullName = createNonRequiredFullName(fullName);
 
 function isUnder65(formData) {
-  return moment().startOf('day').subtract(65, 'years').isBefore(formData.veteranDateOfBirth);
+  return moment().startOf('day').subtract(65, 'years')
+    .isBefore(formData.veteranDateOfBirth);
```
Comment (cvalarida) [inline src/js/pensions/config/form.js]: Is this just because the line is long?

File: src/js/rx/client.js [hunk 2]
```diff
@@ -1,25 +0,0 @@
-// TODO(awong): Convert from webpack common.js require() format to ES6 modules.
```
Comment (cvalarida) [inline src/js/rx/client.js]: Why is this removed?

File: src/js/rx/reducers/errors.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ const initialState = {
   errors: [],
 };
 
-export default function errors(state = initialState, action) {
```
Comment (cvalarida) [inline src/js/rx/reducers/errors.js]: No default param? Is that right?

File: test/common/model/fields.spec.js [hunk 4]
```diff
@@ -1,90 +0,0 @@
-const chai = require('chai');
```
Comment (cvalarida) [inline test/common/model/fields.spec.js]: `src/js/common/model/fields.js` still exists, so should this test really be deleted?

File: test/common/model/schema/definitions.spec.js [hunk 5]
```diff
@@ -1,62 +0,0 @@
-const jsonValidator = require('../../../src/common/schema/validator');
```
Comment (cvalarida) [inline test/common/model/schema/definitions.spec.js]: Is this intentional?

File: test/edu-benefits/1990-rjsf/config/applicantInformation.unit.spec.jsx [hunk 6]
```diff
@@ -1,10 +1,9 @@
 import React from 'react';
 import { expect } from 'chai';
 import ReactTestUtils from 'react-dom/test-utils';
-import { getFormDOM } from '../../../util/schemaform-utils';
 import moment from 'moment';
 
-import { DefinitionTester } from '../../../util/schemaform-utils.jsx';
+import { DefinitionTester, getFormDOM } from '../../../util/schemaform-utils.jsx';
```
Comment (cvalarida) [inline test/edu-benefits/1990-rjsf/config/applicantInformation.unit.spec.jsx]: Probably don't need the extension here.

File: test/edu-benefits/1990-rjsf/config/benefitsEligibility.unit.spec.jsx [hunk 7]
```diff
@@ -1,9 +1,8 @@
 import React from 'react';
 import { expect } from 'chai';
 import ReactTestUtils from 'react-dom/test-utils';
-import { getFormDOM } from '../../../util/schemaform-utils';
 
-import { DefinitionTester } from '../../../util/schemaform-utils.jsx';
+import { DefinitionTester, getFormDOM } from '../../../util/schemaform-utils.jsx';
```
Comment (cvalarida) [inline test/edu-benefits/1990-rjsf/config/benefitsEligibility.unit.spec.jsx]: Or here.

File: test/edu-benefits/1990-rjsf/config/benefitsRelinquishment.unit.spec.jsx [hunk 8]
```diff
@@ -1,11 +1,10 @@
 import React from 'react';
 import { expect } from 'chai';
 import ReactTestUtils from 'react-dom/test-utils';
-import { getFormDOM } from '../../../util/schemaform-utils';
 import sinon from 'sinon';
 import _ from 'lodash/fp';
 
-import { DefinitionTester } from '../../../util/schemaform-utils.jsx';
+import { DefinitionTester, getFormDOM } from '../../../util/schemaform-utils.jsx';
```
Comment (cvalarida) [inline test/edu-benefits/1990-rjsf/config/benefitsRelinquishment.unit.spec.jsx]: ...or here...

File: test/pensions/config/employmentHistory.unit.spec.jsx [hunk 9]
```diff
@@ -1,11 +1,10 @@
 import React from 'react';
 import moment from 'moment';
-import { getFormDOM } from '../../util/schemaform-utils';
 import { expect } from 'chai';
 import sinon from 'sinon';
 import ReactTestUtils from 'react-dom/test-utils';
 
-import { DefinitionTester } from '../../util/schemaform-utils.jsx';
+import { DefinitionTester, getFormDOM } from '../../util/schemaform-utils.jsx';
```
Comment (cvalarida) [inline test/pensions/config/employmentHistory.unit.spec.jsx]: ...ahem... :man_shrugging:

**Review by cvalarida — COMMENTED** [review]
Not sure how some of these weren't already fixed, actually.

That notwithstanding, they mostly look alright, but there were some tests that were deleted entirely (which didn't look right) and some other odd goings on.

Um...also, feel free to ignore the extension comments if you like. I just thought I'd point it out while I saw it.

File: src/js/pensions/config/form.js (hunk 1 above)
Comment (jbalboni) [inline src/js/pensions/config/form.js]: There's a rule about breaking chained calls into multiple lines. It seems weird, but it also only seemed to cause problems in two places, so I left it on.

File: src/js/rx/client.js (hunk 2 above)
Comment (jbalboni) [inline src/js/rx/client.js]: It's never used, as far as I can tell.

File: src/js/rx/reducers/errors.js (hunk 3 above)
Comment (jbalboni) [inline src/js/rx/reducers/errors.js]: The default parameter is unneeded, there's a rule that catches that now.

File: test/common/model/fields.spec.js (hunk 4 above)
Comment (jbalboni) [inline test/common/model/fields.spec.js]: This test is never run. That source file is also never used, so I should delete it, too.

File: test/common/model/schema/definitions.spec.js (hunk 5 above)
Comment (jbalboni) [inline test/common/model/schema/definitions.spec.js]: This test isn't ever run and doesn't make sense anymore, as far as I can tell.

File: test/edu-benefits/1990-rjsf/config/applicantInformation.unit.spec.jsx (hunk 6 above)
Comment (jbalboni) [inline test/edu-benefits/1990-rjsf/config/applicantInformation.unit.spec.jsx]: Nope, but I think we'll end up turning on the rules to prevent using extensions and I can auto-fix them then.

File: test/common/model/fields.spec.js (hunk 4 above)
Comment (jbalboni) [inline test/common/model/fields.spec.js]: Never mind, it is still used. I've updated and moved this test.

**Review by cvalarida — APPROVED** [review]
